components/datetime/src/options/components.rs

```diff
@@ -54,8 +54,14 @@
 //! time. Formatted result should be treated as opaque and displayed to the user as-is,
 //! and it is strongly recommended to never write tests that expect a particular formatted output.
 use super::preferences;
+#[cfg(all(not(feature = "serialize_none"), feature = "serde"))]
+use serde::{Deserialize, Serialize};
 
 #[derive(Debug)]
+#[cfg_attr(
+    all(not(feature = "serialize_none"), feature = "serde"),
+    derive(Clone, Serialize, Deserialize)
```

Comment:
Why make Clone dependent on serde?  Seems like it should be available by default?

Reply:
Mainly to limit the derived features to where they are required, however it would be easy enough to put it on the main derive if that's preferable.

Reply:
I don't understand the advantage of *not* making Clone available by default.  Besides being a generally useful feature to have available, if code isn't actually using the Clone trait, then I think that code won't be built into the binary.  I feel like most public structs should implement Clone unless there's a reason not to.

Reply:
I handled this suggestion, and also added `PartialEq` to where it was needed for the existing test fixtures.